Sources/XiEditor/StyleMap.swift

```diff
@@ -201,17 +208,13 @@ class StyleMapState: UnfairLock {
         return builder.measure()
     }
 
-    func measureWidths(_ args: [[String: AnyObject]]) -> [[Double]] {
+    func measureWidths(_ args: [MeasureWidthParams]) -> [[Double]] {
         Trace.shared.trace("measureWidths", .main, .begin)
         defer { Trace.shared.trace("measureWidths", .main, .end) }
 
-        return args.map({(arg: [String: AnyObject]) -> [Double] in
-            guard let id = arg["id"] as? Int, let strings = arg["strings"] as? [String] else {
-                print("invalid measure_widths request")
-                return []
-            }
-            return strings.map({(s: String) -> Double in measureWidth(id: id, s: s)})
-        })
+        return args.map { p in
```

Comment:
What is `p`? Is it parameter?

Reply:
This is a special type of block/closure that Swift introduced called a trailing closure. If the last parameter to a function call is a block, you can omit the parens and just have the bare `{ ... }` block after the `.map`.  So `p` represents an item in `args` as `map()` is applied.

Reply:
My two cents: any place where we're handling our own data (that is where the client is deserializing something that was generated by core) I'm fine with crashing if the message is malformed.

Reply:
Is this a copy-paste comment? :) I'm not doing error handling or message parsing here. :)

Reply:
oops yep had posted in the wrong place initially, moved it but forgot to delete the original. 👍